Theory/Theory.txt

```diff
@@ -0,0 +1,728 @@
+print(extra_string.isalnum())
+# True
+
+
+isalpha() - returns true if all characters are alphabetical (no special characters or spaces):
+print(string_two.isalpha())
+# False
+
+
+isdigit() - returns true if all characters are only digits:
+print(extra_string.isdigit())
+# False
+
+
+islower() - returns true if all characters are lowercase:
+print(my_string.islower())
+# True
+
+
+isnumeric() - returns true if all characters are numeric, no special characters, so floats also return false:
+print(number_string.isnumeric())
+# False
+
+
+isspace() - returns true if ALL characters are spaces:
+print(my_string.isspace())
+# False
+
+
+istitle() - returns true if the rules of a title are met, so all words have the first letter uppercase:
+print(string_two.istitle())
+# True
+
+
+isupper() - returns true if all characters are uppercase (numbers are ignored, so can be present):
+print(extra_string.isupper())
+# True
+
+
+join() - takes items from an iterable - list, dictionary, tuple - and forms a string:
+a_list = ['I\'m', 'still', 'talking', 'about', 'TACOS']
+print(' '.join(a_list))
+# I'm still talking about TACOS
+
+
+lower() - converts all characters to lowercase:
+print(extra_string.lower())
+# becausetheyareawesome123
+
+
+lstrip() - removes leading (left) spaces:
+print(spaces_string.lstrip())
+# 'Hello, Tacos?               '
+
+
+replace() - replaces a word or phrase with a defined character. It takes two arguments, a character to be replaced and
+a character to replace it with:
+print(spaces_string.replace(' ', '*'))
+# ***Hello,*Tacos?***************
+
+
+rsplit() - splits a string to a list starting from the right (end) and takes a defined separator for the break:
+print(extra_string.rsplit('1'))
+ # ['BECAUSETHEYAREAWESOME', '23']
+
+
+rstrip() - removes trailing (right) spaces:
+print(spaces_string.rstrip())
+#   Hello, Tacos?               '
+
+
+split() - splits the string to a list of each word:
+print(my_string.split())
+# ['hey', 'if', 'tacos', 'can', 'totally', 'fall', 'apart', 'and', 'still', 'be', 'amazing', 'so', 'can', 'you']
+
+
+splitlines() - splits the string to a list at the line breaks - so each line is an item:
+print(my_string.splitlines())
+# ['hey if tacos can totally fall apart and still be amazing so can you']
+
+
+startswith() - returns true if string starts with the specified value (again is case-sensitive):
+print(string_two.startswith('i'))
+# False
+
+
+strip() - returns a version trimmed of all white space:
+print(spaces_string.strip())
+# Hello, Tacos?
+
+
+swapcase() - converts lowercase characters to uppercase and vice-versa:
+print(string_two.swapcase())
+# i lOVE tACOS
+
+
+title() - converts string into title style, the first character of each word to uppercase:
+print(my_string.title())
+# Hey If Tacos Can Totally Fall Apart And Still Be Amazing So Can You
+
+
+upper() - converts all characters to uppercase:
+print(last_string.upper())
+# I'M SORRY IF I'VE MADE YOU HUNGRY
+
```

Comment:
29/29 - Haha, nice use of tacos indeed you. successfully made me want to have some tacos